server/src/gateway/providers/cursor.action.service.ts

```diff
@@ -0,0 +1,40 @@
+// src/gateway/providers/cursor.action.service.ts
+
+import { Injectable, Logger } from '@nestjs/common';
+import { GwSocketWithTarget } from '../../shared/interfaces/auth/GwSocket';
+import { CursorMoveData } from '../dto/cursor.data';
+
+@Injectable()
+export class CursorActionService {
+  private readonly logger = new Logger(CursorActionService.name);
+
+  handleCursorMove(
+    client: GwSocketWithTarget,
+    data: CursorMoveData,
+  ): void {
+    const user  = client.data.user;
+    const { x, y, color } = data;
+    const slideId = client.data.user.targetSlide.slideId;
+
+    if (!slideId) {
```

Comment:
it cant be such situation - for joined-board users a slideId is always defined

Reply:
Actually, i encountered a situation when this if was true, i think to better leave this as it is (i don't know what exactly happened, i just encountered once)

Reply:
Yeah, I think, especially during development, it might be better to notify ourselves when something goes seriously wrong, like when we can't connect to the slide but still emit mouse movement. (That is what happenned before, when one property was wrongly named)

Reply:
then it should be handled in the guard not in the service - i guess there is some invalid handling

Reply:
and this method should not be responsible for that

Reply:
What about object.action.service.ts? There are plenty of these types of checkers.